readthedocs/projects/forms.py

```diff
@@ -272,6 +273,69 @@ def __init__(self, *args, **kwargs):
         else:
             self.fields['default_version'].widget.attrs['readonly'] = True
 
+        self.setup_external_builds_option()
+
+    def setup_external_builds_option(self):
+        """Disable the external builds option if the project doesn't meet the requirements."""
+        msg = None
+        url = reverse("projects_integrations", args=[self.instance.slug])
+        integrations = list(self.instance.integrations.all())
+        if not self.has_supported_integration(integrations):
+            msg = _(
+                "To build from pull requests you need a "
+                'GitHub or GitLab <a href="{url}">integration</a>.'
+            )
+        elif not self.can_build_external_versions(integrations):
+            # If there is only one integration, link directly to it.
+            if len(integrations) == 1:
+                url = reverse(
+                    "projects_integrations_detail",
+                    args=[self.instance.slug, integrations[0].pk],
+                )
+            msg = _(
+                "To build from pull requests your repository's webhook "
+                "needs to send pull request events. "
+                f'Try to <a href="{url}">resync your integration</a>.'
+            )
+
+        if msg:
+            field = self.fields["external_builds_enabled"]
+            field.disabled = True
+            field.help_text = msg + " " + field.help_text
+
+    def has_supported_integration(self, integrations):
+        supported_types = {Integration.GITHUB_WEBHOOK, Integration.GITLAB_WEBHOOK}
+        for integration in integrations:
+            if integration.integration_type in supported_types:
+                return True
+        return False
+
+    def can_build_external_versions(self, integrations):
+        """
+        Check if external versions can be enabled for this project.
+
+        A project can build external versions if:
+
+        - They are using GitHub or GitLab.
+        - The repository's webhook is setup to send pull request events.
+
+        If the integration's provider data isn't set,
+        it could mean that the user created the integration manually,
+        and doesn't have an account connected.
+        So we don't know for sure if the webhook is sending pull request events.
```

Comment:
We should mark this case as "can't build external versions". Otherwise, some people will enable it and it won't work. It's better to let people enable the feature if we are 100% sure it will work.

Besides, these cases will be hard to track and debug when receiving a support request like "My PRs are not building".

Reply:
We should still support those cases, these are users using an enterprise instance of GitHub or users that don't want to give full permissions to RTD by connecting their accounts.

Reply:
They won't receive the commit status, but at least the PR will build.

Reply:
> They won't receive the commit status, but at least the PR will build.

Seems we should at least mention this in the UI as help text, then.